scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -159,9 +161,25 @@
                      convert_to=int, default=20)
 
 define_config_option( __monitor__, 'k8s_parse_json',
-                      'Optional (defaults to True). If True, the log files will be parsed as json before uploading to the server '
-                      'to extract log and timestamp fields.  If False, the raw json will be uploaded to Scalyr.',
-                      convert_to=bool, default=True, env_aware=True)
+                      'Deprecated, please use `k8s_parse_format`. If set, and True, then this flag will override the `k8s_parse_format` to `auto`. '
+                      'If set and False, then this flag will override the `k8s_parse_format` to `raw`.',
+                      convert_to=bool, default=None)
+
+define_config_option( __monitor__, 'k8s_parse_format',
+                      'Optional (defaults to `auto`). Valid values are: `auto`, `json`, `cri` and `raw`. '
+                      'If `auto`, the monitor will try to detect the format of the raw log files, '
+                      'e.g. `json` or `cri`.  Log files will be parsed in this format before uploading to the server '
+                      'to extract log and timestamp fields.  If `raw`, the raw contents of the log will be uploaded to Scalyr.',
```

Comment:
As discussed, I propose we add a disclaimer warning users not to set this unless they know what they are doing.  For example:
"(Note: An incorrect setting will cause parsing to fail so please leave this as `auto` if in doubt.)"

Reply:
Done.  But modified slightly to add what will happen if parsing fails:

> Note: An incorrect setting can cause parsing to fail which will result in raw logs being uploaded to Scalyr, so please leave this as `auto` if in doubt.

Reply:
👍